src/main/java/io/gravitee/reporter/common/bulk/sender/BulkSender.java

```diff
@@ -0,0 +1,28 @@
+/*
+ * Copyright © 2015 The Gravitee team (http://gravitee.io)
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package io.gravitee.reporter.common.bulk.sender;
+
+import io.gravitee.common.service.Service;
+import io.gravitee.reporter.common.bulk.compressor.CompressedBulk;
+import io.reactivex.rxjava3.core.Completable;
+
+/**
+ * @author Guillaume LAMIRAND (guillaume.lamirand at graviteesource.com)
+ * @author GraviteeSource Team
+ */
+public interface BulkSender extends Service<BulkSender> {
+  Completable send(CompressedBulk bulk);
```

Comment:
Maybe add some documentation to explain that it is the responsability of the implementation to determine if a NonRetryableException should be used or not and the processor will retry based on the kind of exception. Wdyt ?